services/wallet/cryptocompare.go

```diff
@@ -32,6 +32,22 @@ type MarketCoinValues struct {
 	CHANGE24HOUR    string `json:"CHANGE24HOUR"`
 }
 
+type TokenHistoricalPairs struct {
+	Timestamp int64   `json:"time"`
+	Value     float64 `json:"close"`
+}
+
+type HistoricalValuesContainer struct {
+	Aggregated     bool                   `json:"Aggregated"`
+	TimeFrom       int64                  `json:"TimeFrom"`
+	TimeTo         int64                  `json:"TimeTo"`
+	HistoricalData []TokenHistoricalPairs `json:"Data"`
+}
+
+type HistoricalValuesData struct {
+	Data HistoricalValuesContainer `json:"Data"`
```

Comment:
why do u need a struct containing only one attribute?

Reply:
Looks like this is needed for marshaling. Looks like the the response of the http requests returns Json that has a `Data` field.

Reply:
Exactly! needed it for unmarshalling